server/controllers/userController.js

```diff
@@ -0,0 +1,163 @@
+import crypto from 'crypto';
+import userModel from '../models/users';
+import encypt from '../helpers/encrypt';
+import emailModule from '../utilities/EmailModule';
+
+const { encryptPassword, decryptPassword, generateToken } = encypt;
+
+/**
+ *
+ *
+ * @export
+ * @class UsersController
+ */
+export default class UsersController {
+  /**
+   * Signup middleware - Create User Account
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {object} Object containing token to the user
+   * @memberof UsersController
+   */
+  static async signUp(req, res, next) {
+    try {
+      const data = req.body;
+      const { password } = data;
+      data.password = await encryptPassword(password);
```

Comment:
you already destructure password on line 27.. why use data.password on line 28 instead of password

Reply:
@akpante3 , Password is a constant and using password on line 28 will return an error because it'll be trying to reassign a constant. 

Also, I think the idea is to rewrite the password property in the data object with the hashed password that will be saved to the DB

Reply:
Thank you @LukasChiama for the clearification. @akpante3 if you still need more explanations on these, i'll suggest you reach out to @Xwebyna since he actually implemented that.